src/unity/toolkits/activity_classification/sequence_iterator.cpp

```diff
@@ -24,27 +24,30 @@ static std::map<std::string,size_t> generate_column_index_map(const std::vector<
 
 /**
  *  Find the statistical mode (majority value) of a given vector.
- *  Based on https://en.wikipedia.org/wiki/Boyer–Moore_majority_vote_algorithm
  *
  * \param[in] input_vec a vector for which the mode will be calculated
 
  * \return    The most frequent value within the given vector.
  */
-static double vec_majority_value(const flex_vec& input_vec) {
-    size_t counter = 0;
-    double candidate = 0.0;
+
+static double vec_mode(const flex_vec& input_vec) {
+    std::vector<int> histogram;
     for (size_t i = 0; i < input_vec.size(); ++i) {
-        double value = input_vec[i];
-        if (counter == 0) {
-            candidate = value;
-            counter = 1;
-        } else if (candidate == value) {
-            counter++;
-        } else {
-            counter--;
+        int value = static_cast<int>(input_vec[i]);
+
+        // Each value should be in the index of a class label
+        assert(static_cast<double>(static_cast<int>(input_vec[i])) == input_vec[i]);
+
+        if(histogram.size() < (value + 1)){
+          histogram.resize(value + 1);
         }
+
+        histogram[value]++;
     }
-    return candidate;
+
+    auto majority = std::max_element(histogram.begin(), histogram.end());
+    
+    return (majority - histogram.begin());
```

Comment:
What is this subtraction ...? I would have expected something like `return majority;` here. Or if majority is an iterator, `return *majority;`.

Reply:
It's the offset. std::max_element returns the iterator. Subtracting the iterator at the max element with the iterator in the beginning gives the offset. I'm changing it to std::distance to make it more evident what I'm doing.

Reply:
Since it's a random access iterator, the std::distance function uses the `-` operator

Reply:
I think we can clarify things by changing the variable name from `majority` to something like `largest_count`. Then it's more clear that we can't return `*largest_count` but must instead compute the corresponding index into the vector (getting the argmax instead of the max). Might also be worth a comment